sktime/annotation/ggs.py

```diff
@@ -34,19 +43,46 @@ class GGS:
     """
     Greedy Gaussian Segmentation.
 
+    The method apprixmates solutions for the problem of breaking a
+    multivariate time series into segments, where the data in each segment
+    could be modeled as independent samples from a multivariate Gaussian
+    distribution. It uses a dynamic programming search algorithm with
+    a heuristic that allows finding approximate solution in linear time with
+    respect to the data length and always yields locally optimal choice.
+
+    Greedy Gaussian Segmentation (GGS) fits a segmented gaussian model (SGM)
+    to the data by computing the appriximate solution to the combinatorial
+    problem of finding the approximate covariance-regularized  maximum
+    log-likelihood for fixed number of change points and a reagularization
+    strength. It follows an interative procedure
+    where a new breakpoint is added and then adjusting all breakpoints to
+    (approximately) maximize the objective. It is similar to the top-down
+    search used in other change point detection problems.
+
     Parameters
     ----------
     k_max: int, default=10
-        Maximum number of change points to find
+        Maximum number of change points to find. The number of segments is thus k+1.
     lamb: : float, default=1.0
-        Regularization parameter lambda
+        Regularization parameter lambda (>= 0), which controls the amount of
```

Comment:
Any chance to get a bit more detail here?  - I don't think you need the full equation but a quick sentence on what affect increasing/decreasing lamba has on the fit would be nice without needing to dive into the paper.

Reply:
Added some extra info. The original paper does not go into much depth about this and as with any hyperparameter it's a question of CV or intuition to choose good values.